### PR TITLE
Pull request to run CI for #72556

### PR DESCRIPTION
**Pull request by kstant0725**
This PR moves the Dockerfile conda dependencies into a requirements-ci.txt (and begins the requirements file for other parts of CI as well).  Packages are listed alphabetically in the requirements-ci.txt.  Uncommented packages before the mkl package have been tested and confirmed to work on all platforms.  Commented out packages before mkl have broken at least one platform and so have been comment out.  There appears to be some randomness with certain platforms not passing tests so it might be good to run a number of tests for the same configuration to confirm if it is indeed these commented out packages that cause the errors.

Remaining is to test all commented out packages to ensure they work on all platforms.  This will likely involve repeat runs of the same configurations to ensure it is indeed the packages that break the platforms and not random errors.

This PR makes progress on task #72556

### CODE REVIEW

File: .circleci/docker/requirements-ci.txt [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+#test that import:
+
+scipy==1.6.3
+#Description: scientific python
+#Pinned versions: 1.6.3
+#test that import: test_unary_ufuncs.py, test_torch.py,test_tensor_creation_ops.py
+#test_spectral_ops.py, test_sparse_csr.py, test_reductions.py,test_nn.py
+#test_linalg.py, test_binary_ufuncs.py
+
+#six
```
Comment (janeyx99) [inline .circleci/docker/requirements-ci.txt:184]: We should be able to get rid of this now that we don't support python 2 anymore

File: .circleci/docker/requirements-ci.txt [hunk 2]
```diff
@@ -0,0 +1,214 @@
+# Python dependencies required for unit tests
+
+#awscli
+#Description: AWS command line interface
+#Pinned versions: 1.6
```
Comment (janeyx99) [inline .circleci/docker/requirements-ci.txt:5]: Does this mean that it's pinned elsewhere? 
Why do we not already write 
awscli==1.6
if there's only one pin?

Comment (kstant0725) [reply .circleci/docker/requirements-ci.txt:5]: There is only one pin, 1.6.  The awscli is currently commented out.  I am still trying to get the install to work without errors on some platforms

File: .circleci/docker/requirements-ci.txt (hunk 1 above)
Comment (kstant0725) [reply .circleci/docker/requirements-ci.txt:184]: OK ill remove six from the requirements-ci.txt

**Review by janeyx99 — COMMENTED** [review]
This PR looks like it moves the Dockerfile conda dependencies into a requirements.txt (and begins the requirements file for other parts of CI). Could you rewrite the title and body to describe the progress here? 

Could you also link this in the issue and describe its current progress/what is remaining?

File: .circleci/docker/common/install_conda.sh [hunk 3]
```diff
@@ -97,17 +97,19 @@ if [ -n "$ANACONDA_PYTHON_VERSION" ]; then
...
-    scikit-image \
-    psutil \
-    unittest-xml-reporting \
-    boto3==1.16.34 \
-    hypothesis==4.53.2 \
-    expecttest==0.1.3 \
-    mypy==0.812 \
-    tb-nightly \
-    librosa>=0.6.2
+  # as_jenkins pip install --progress-bar off boto3==1.16.34 \
```
Comment (janeyx99) [inline .circleci/docker/common/install_conda.sh:100]: I'm guessing you will remove this commented out part? ;)

Comment (kstant0725) [reply .circleci/docker/common/install_conda.sh:100]: Yes, i just removed it

**Comment by kstant0725** [comment]
> 

Updatd the title

**Review by janeyx99 — COMMENTED** [review]
The mypy failures are due to this code:
https://github.com/pytorch/pytorch/blob/1022443168b5fad55bbd03d087abf574c9d2e9df/mypy_plugins/check_mypy_version.py#L12
The file there should be the added requirements file

**Review by janeyx99 — APPROVED** [review]

**Pull request closed without merging**